packages/cli/src/ceramic-daemon.ts

```diff
@@ -524,36 +517,111 @@ export class CeramicDaemon {
     }
   }
 
-  async addModelsToIndex(req: Request, res: Response): Promise<void> {
-    const { modelIDStrings, error } = this._validateModelIDStrings(req.body.models)
-    if (error) {
-      res.status(422).json({ error: error })
+  private async _parseDidJWSAuthHeader(
+    authHeader: string | undefined
+  ): Promise<{ kid: string, requestPath: string, timestamp: number, forModels: Array<string> }> {
+    const jwsString = authHeader.split("Authorization: Basic ")[1]
+    const result = await this.ceramic.did.verifyJWS(jwsString)
+    return {
+      kid: result.kid,
+      requestPath: result.payload.requestPath,
+      timestamp: result.payload.timestamp,
+      forModels: result.payload.requestBody ? result.payload.requestBody.models : undefined,
+    }
+  }
+
+  private _compareStringArrays(left: Array<string>, right: Array<string>): boolean {
+    return left.length === right.length && left.every((value, index) => value === right[index])
+  }
+
+  private async _validateDIDJWSAuthHeader(
+    basePath: string,
+    authHeader: string | undefined,
+    forModels?: Array<string>
+  ): Promise<{ kid?: string, error?: string }> {
+    const errorResult = { status: 'error', error: 'Missing or invalid authorization signature' }
+    let parsedJWS
+    try {
+      parsedJWS = await this._parseDidJWSAuthHeader(authHeader)
+    } catch (e) {
+      console.error(e)
+      return errorResult
+    }
+    const fiveMin = 1000 * 60 * 5 // 5 min
+
+    if (
+      parsedJWS.requestPath === basePath &&
+      parsedJWS.timestamp > (new Date().getTime() - fiveMin)
+    ) {
+      if (forModels) {
+        if (!parsedJWS.forModels || !this._compareStringArrays(forModels.sort(), parsedJWS.forModels.sort())) {
+          return errorResult
+        }
+      } else if (parsedJWS.forModels) {
+        return errorResult
+      }
+      return { kid: parsedJWS.kid }
     } else {
-      await this.ceramic.admin.startIndexingModels(modelIDStrings.map( modelIDString => StreamID.fromString(modelIDString) ))
-      res.status(200).json({ result: 'success' })
+      return errorResult
     }
   }
 
-  async removeModelsFromIndex(req: Request, res: Response): Promise<void> {
-    const { modelIDStrings, error } = this._validateModelIDStrings(req.body.models)
-    if (error) {
-      res.status(422).json({ error: error })
+  private async _processAdminModelsMutationRequest(
+    req: Request,
+    res: Response,
+    successCallback: (actingDid: string, modelIDs: Array<StreamID>) => Promise<void>
+  ): Promise<void> {
+    const modelsValidation = this._validateModelIDStrings(req.body.models)
+    if (modelsValidation.error) {
+      res.status(422).json({ error: modelsValidation.error })
```

Comment:
is there a library with named constants for these http codes so they are more semantically meaningful to a reader?